### PR TITLE
feat(eks): upgrade latest kubertenes version to 1.14

### PR DESCRIPTION
**Pull request by nmussy**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nmussy** [comment]
EKS added Kubernetes 1.14 support [earlier this month](https://aws.amazon.com/about-aws/whats-new/2019/09/amazon-eks-now-supports-kubernetes-version-1-14/).

I've only bumped the constant without adding the AMIs to the static table, waiting for SSM support to be merged (#4156)

**Comment by nmussy** [comment]
@eladb Is too late to block it now? I wanted to wait until #4156 was merged

**Comment by eladb** [comment]
Let’s see...

**Comment by nmussy** [comment]
I should have made a draft, sorry about that. I'm making the changes to #4156 now anyway

File: packages/@aws-cdk/aws-eks/test/integ.eks-cluster.defaults.expected.json [hunk 1]
```diff
@@ -1051,7 +1051,7 @@
       "Type": "AWS::AutoScaling::LaunchConfiguration",
       "Properties": {
         "ImageId": {
-          "Ref": "SsmParameterValueawsserviceeksoptimizedami113amazonlinux2recommendedimageidC96584B6F00A464EAD1953AFF4B05118Parameter"
+          "Ref": "SsmParameterValueawsserviceeksoptimizedami114amazonlinux2recommendedimageidC96584B6F00A464EAD1953AFF4B05118Parameter"
```
Comment (nmussy) [inline packages/@aws-cdk/aws-eks/test/integ.eks-cluster.defaults.expected.json]: @eladb I kind of want to add a small shell script to help the next contributor bump the latest kube version, what do you think?

Comment (nmussy) [reply packages/@aws-cdk/aws-eks/test/integ.eks-cluster.defaults.expected.json]: I went ahead and did it, tell me if you want to rename the script

File: scripts/kube_bump.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
```
Comment (eladb) [inline scripts/kube_bump.sh]: Please move this under `packages/@aws-cdk/aws-eks/scripts`

Comment (nmussy) [reply scripts/kube_bump.sh]: Done

**Pull request merged**